From 9f0d8a6e7444dd59729bac2c24717ea72aea5080 Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@gnome.org>
Date: Thu, 21 Jul 2016 20:48:12 -0500
Subject: [PATCH] Revert "Don't call XInitThreads in Wayland"

This reverts commit a9ede294c57bc5738be3c33cba4ef88b9a7d4a0e.

It causes empathy-call to crash when run in X11, and fixes absolutely
nothing. See the bug for details.

Thanks to Diane Trout for investigating and preparing a substantially-
identical patch.

https://bugzilla.gnome.org/show_bug.cgi?id=767516
---
 src/empathy-call.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/empathy-call.c b/src/empathy-call.c
index 8821bf3..1ac8a79 100644
--- a/src/empathy-call.c
+++ b/src/empathy-call.c
@@ -28,7 +28,6 @@
 #include <tp-account-widgets/tpaw-utils.h>
 
 #ifdef CLUTTER_WINDOWING_X11
-#include <gdk/gdkx.h>
 #include <X11/Xlib.h>
 #endif
 
@@ -216,11 +215,9 @@ main (int argc,
   g_setenv ("GST_DEBUG_DUMP_DOT_DIR", g_get_tmp_dir (), FALSE);
 
 #ifdef GDK_WINDOWING_X11
-  if (GDK_IS_X11_DISPLAY (gdk_display_get_default ())) {
-    /* We can't call clutter_gst_init() before gtk_clutter_init(), so no choice
-     * but to intiialise X11 threading ourself */
-    XInitThreads ();
-  }
+  /* We can't call clutter_gst_init() before gtk_clutter_init(), so no choice
+   * but to intiialise X11 threading ourself */
+  XInitThreads ();
 #endif
 
   optcontext = g_option_context_new (N_("- Empathy Audio/Video Client"));
